src/hydra/HydraAdmin.js

```diff
@@ -7,34 +7,57 @@ import restClient from './hydraClient';
 class HydraAdmin extends Component {
   static defaultProps = {
     apiDocumentationParser,
+    customRoutes: [],
     restClient,
   };
 
   static propTypes = {
     apiDocumentationParser: PropTypes.func,
+    customRoutes: PropTypes.array,
     entrypoint: PropTypes.string.isRequired,
     restClient: PropTypes.func,
   };
 
   state = {
     api: null,
+    customRoutes: [],
+    hasError: false,
+    loaded: false,
   };
 
   componentDidMount() {
-    this.props
-      .apiDocumentationParser(this.props.entrypoint)
-      .then(api => this.setState({api}));
+    this.props.apiDocumentationParser(this.props.entrypoint).then(
+      ({api, customRoutes = []}) =>
+        this.setState({
+          api,
+          customRoutes,
+          hasError: false,
+          loaded: true,
+        }),
+      ({api, customRoutes = []}) =>
+        this.setState({
+          api,
+          customRoutes,
+          hasError: true,
+          loaded: true,
+        }),
+    );
   }
 
   render() {
-    if (null === this.state.api) {
+    if (false === this.state.loaded) {
       return <span>Loading...</span>;
     }
 
+    if (true === this.state.hasError) {
+      return <span>Impossible to retrieve API documentation.</span>;
```

Comment:
maybe this could be a key instead of directly english ?

Reply:
Allowing to pass props as suggested by @Gregcop1 looks better to me.